src/humanloop/eval_utils.py

```diff
@@ -0,0 +1,543 @@
+        case _:
+            raise NotImplementedError(f"Unsupported File version: {type_}")
+
+
+def get_score_from_evaluator_stat(stat: NumericStats | BooleanStats) -> float | None:
+    """Get the score from an Evaluator Stat."""
+    score = None
+    match stat:
+        case BooleanStats():
+            if stat.total_logs:
+                score = round(stat.num_true / stat.total_logs, 2)
+        case NumericStats():
+            score = round(stat.mean, 2)
+        case _:
+            raise ValueError("Invalid Evaluator Stat type.")
+    return score
+
+
+def _progress_bar(total: int, progress: int):
+    """Simple progress bar for CLI with ETA."""
+
+    if total <= 0:
+        total = 1
+
+    if not hasattr(_progress_bar, 'start_time'):
+        _progress_bar.start_time = time.time()
+
+    bar_length = 40
+    block = int(round(bar_length * progress / total))
+    bar = "#" * block + '-' * (bar_length - block)
+
+    percentage = (progress / total) * 100
+    elapsed_time = time.time() - _progress_bar.start_time
+    time_per_item = elapsed_time / progress if progress > 0 else 0
+    eta = (total - progress) * time_per_item
+
+    progress_display = f"\r[{bar}] {progress}/{total}"
+    progress_display += f" ({percentage:.2f}%)"
+
+    if progress < total:
+        progress_display += f" | ETA: {int(eta)}s"
+    else:
+        progress_display += " | DONE"
+        _progress_bar.start_time = None
+
+    sys.stdout.write(progress_display)
+    sys.stdout.flush()
+
+    if progress >= total:
+        sys.stdout.write("\n")
+
+
+def get_evaluator_stats_by_path(
+    stat: VersionStatsResponse, evaluation: EvaluationResponse
+) -> dict[str, NumericStats | BooleanStats]:
+    """Get the Evaluator stats by path."""
+    # TODO: Update the API so this is not necessary
+    evaluators_by_id = {
+        evaluator.version.version_id: evaluator for evaluator in evaluation.evaluators
+    }
+    evaluator_stats_by_path = {
+        evaluators_by_id[evaluator_stat.evaluator_version_id].version.path: evaluator_stat
+        for evaluator_stat in stat.evaluator_version_stats
+    }
+    return evaluator_stats_by_path
+
+
+def check_evaluation_threshold(
+    evaluation: EvaluationResponse,
+    stats: EvaluationStats,
+    evaluator_path: str,
+    threshold: float,
+) -> bool:
+    """Checks if the latest version has an average Evaluator result above a threshold."""
+    # TODO: Update the API so this is not necessary
+    evaluator_stats_by_path = get_evaluator_stats_by_path(
+        stat=stats.version_stats[-1],
+        evaluation=evaluation
+    )
+    if evaluator_path in evaluator_stats_by_path:
+        evaluator_stat = evaluator_stats_by_path[evaluator_path]
+        score = get_score_from_evaluator_stat(stat=evaluator_stat)
+        if score >= threshold:
+            print(
+                f"{GREEN}✅ Latest eval [{score}] above threshold [{threshold}] for evaluator {evaluator_path}.{RESET}"
```

Comment:
can we make the text consistent with the below threshold print? e.g. latest eval vs latest score